salt/_auth/kubernetes_rbac.py

```diff
@@ -114,14 +114,54 @@ def _groups_basic(kubeconfig, username, token):
     'groups': _groups_basic,
 }
 
+
 @_log_exceptions
 def _auth_bearer(kubeconfig, username, token):
     return _check_k8s_creds(kubeconfig, 'Bearer {}'.format(token))
 
+
+@_log_exceptions
+def _groups_bearer(kubeconfig, username, token):
+    kubeconfig.api_key = {
+        'authorization': token,
+    }
+    kubeconfig.api_key_prefix = {
+        'authorization': 'Bearer',
+    }
+    kubeconfig.username = username
+    kubeconfig.password = None
+    kubeconfig.cert_file = None
+    kubeconfig.key_file = None
+
+    client = kubernetes.client.ApiClient(configuration=kubeconfig)
+
+    authz_api = kubernetes.client.AuthorizationV1Api(api_client=client)
+
+    groups = set()
+
+    result = authz_api.create_self_subject_access_review(
+        body=kubernetes.client.V1SelfSubjectAccessReview(
+            spec=kubernetes.client.V1SelfSubjectAccessReviewSpec(
+                resource_attributes=kubernetes.client.V1ResourceAttributes(
+                    resource='nodes',
+                    verb='*',
+                ),
+            ),
+        ),
+    )
```

Comment:
Could be worth factorizing with the code in `_groups_basic`

Reply:
Since this one should change in the futur (and I hope before 2.5.0 release), to handle different roles I think it's not really needed

Reply:
Disagree. The "node-admins" group for Salt will still exist I guess, and even if the way to know if a user is in that group changes in the future, there's common logic between Basic and Bearer tokens.

Reply:
Agree with @gdemonet. The mapping of K8s rights to Salt groups is intricate, and should be *the same* whether you're using Basic auth, Bearer tokens or whatever. The only difference is the mangling of the `kubeconfig` object (which is an ugly hack anyway).